dstk-infra/apollo/src/graphql/model-version/modelVersionQueries.ts

```diff
@@ -16,3 +16,20 @@ export const ListModelVersions = extendType({
         });
     },
 });
+
+export const ListAllModelVersions = extendType({
+    type: 'Query',
+    definition(t) {
+        t.nonNull.list.field('listAllModelVersions', {
+            type: MLModelVersion,
+            async resolve(_root, _args, _ctx) {
+                const result = await ObjectionMLModelVersion.query().where(
+                    'numericVersion',
+                    '=',
+                    ObjectionMLModelVersion.query().max('numericVersion'),
```

Comment:
Oh yeah, good catch. Think I need to add a `.groupBy('modelId')`

Reply:
I'm not sure that would solve this. I think that whatever solution you come to will end up having something like a subquery with a RANK or similar; there's not (unless I'm having a major brain fart which, I mean, is always a legit possiblity) a straightfoward way to do this operation without the SQL getting a little weird. (cf. https://stackoverflow.com/a/2657495)

If you want to make life a little easier on yourself, we could I think take one of 2 routes:

  1. Just run a simple query for each model to get its max model version. This has the downside of being many small queries.
  2. Add a `currentModelVersion` column to `registry.models` that's a FK reference to whatever the most recent version for that model is at the time. This would require some small edits to the `CreateModelVersion` mutation, but that's like a 2-line change. From there, you could just call it a day with

```typescript
const result = await ObjectionMLModel.relatedQuery('currentModelVersion') as ObjectionMLModelVersion;
```

Reply:
I'll get back to you on this. I'm getting a `readarray: command not found` when trying to apply the postgres patches. I don't really know about shells and what not, so I'm going to procrastinate by getiing my Emergency Domino's pizza and looking at Christmas lights 😂

I'll give this another look tonight or tomorrow morning.

Reply:
Ohhhh probably your shell is zsh and not bash; I forgot `readarray` doesn't port over to zsh cleanly. It should have picked up the hash bang at the top of the file though... Weird

In the meanwhile, try `bash ./bin/storage upgrade`

Reply:
Yeah I tried that already still not recognizing the `readarray` command. I'm going to try upgrading bash and see if that works.

Reply:
Upgrading bash worked!

Reply:
Ahh yeah Apple ships computers with older versions of bash, that could be it. #74 should add ZSH compatibility, which is the recommended shell for macs nowadays, so that's also an option if you just want to make that switch locally